### ISSUE TITLE
generateSignInWithEmailLink(...) in Admin Auth DSK

### ISSUE THREAD

**Issue by mkilmanas** [kreait/firebase-php#359]
**Is your feature request related to a problem? Please describe.**
We are implementing Firebase authentication on our project, which uses Javascript Client SDK for direct user interaction, but we want to use PHP Admin SDK for generating and sending the login links by email directly to the user [(Basically this thing)](https://firebase.google.com/docs/auth/admin/email-action-links#generate_email_link_for_sign-in) But I see this particular method is not (yet?) implemented in this lib.

**Describe the solution you'd like**
I would love to see the `generateSignInWithEmailLink(...)` method added to the Auth SDK

**Describe alternatives you've considered**
Using the Admin SDK from another language?... but that seems like a huge overkill to add a new language to the "zoo" just for the sake of this single call.

**Additional context**
[Here](https://github.com/kreait/firebase-php/issues/312#issuecomment-520166635) it was mentioned that it is likely to be added in the future release.
With some guidance on where to start and how to properly implement this I don't mind getting my hands dirty and I could contribute in the form of PR(s).
But I'm rather fresh to Firebase terminology and I'm not quite fully comprehending the structure/architecture of this library, so might need quite some help.

**Comment by jeromegamez** [comment]
Thanks for letting me know/reminding me! I‘ve had a Look at the other SDKs and the internals have changed a little since I last touched the component, so it will take me a bit to apply this to the library - but know that this is now on the top of my todo list.

**Comment by mkilmanas** [comment]
Glad to hear! Let me know if I can help in any way. Looking forward to this

**Comment by jeromegamez** [comment]
@mkilmanas I just created a draft PR at #360 which should include the basic functionality, would you like to give it a try and see if it works for you so far? There is still a lot missing, but I believe I'm onto something ^^.

**Comment by mkilmanas** [comment]
@jeromegamez looks great! I've tried it out and seems to work nicely. 
All but one thing - is it currently respecting the ActionCodeSettings parameter? From what I found in the [Firebase Docs](https://firebase.google.com/docs/auth/admin/email-action-links#initialize_actioncodesettings) the 'url' parameter should mean the 'continueUrl', i.e. where to redirect upon verification (or did I misinterpret that?), but currently I get "Error encountered / Continue URL is required for email sign-in!". It only works if I manually append `'&continueUrl=' . urlencode($continueUrl);` at the end of the retrieved url. Is that expected? Or is that To-Do?
Anyway, thanks a bunch for such a quick turnaround 👍
